packages/mailgun/src/rest/v1/emails/emails-api.ts

```diff
@@ -0,0 +1,193 @@
+    this.client = this.getSinchClient();
+    const getParams = this.client.extractQueryParams<SendMimeEmailRequest>(request, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      Accept: 'application/json',
+    };
+    const body: RequestBody = transformSendMimeEmailRequestIntoApiRequestBody(request);
+    const basePathUrl = `${this.client.apiClientOptions.hostname}/v3/${domainName}/messages.mime`;
+
+    const requestOptions = await this.client.prepareOptions(basePathUrl, 'POST', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.hostname, requestOptions.queryParams);
+
+    const apiResponse = await this.client.processCall<SendEmailResponseFromApi>({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'sendMimeEmail',
+    });
+
+    return transformSendEmailResponseIntoClientResponse(apiResponse);
+  }
+
+  /**
+   * Retrieve a stored email
+   * Event(s) created from sending an email with Mailgun will contain a `storage.key` to use to retrieve the email.
+   * @param { string } domainName - Domain name that was used to send the email
+   * @param { string } storageKey - Storage key from the emails associated events
+   */
+  public async getStoredEmail(domainName: string, storageKey: string): Promise<GetStoredEmailResponse> {
+    this.client = this.getSinchClient();
+    const getParams = {};
+    const headers: { [key: string]: string | undefined } = {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    };
+    const body: RequestBody = '';
+    const basePathUrl = `${this.client.apiClientOptions.hostname}/v3/domains/${domainName}/messages/${storageKey}`;
+
+    const requestOptions = await this.client.prepareOptions(basePathUrl, 'GET', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.hostname, requestOptions.queryParams);
+
+    const apiResponse = await this.client.processCall<GetStoredEmailResponseFromApi>({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'getEmail',
+    });
+
+    return transformGetEmailResponseIntoClientResponse(apiResponse);
+  }
+
+  /**
+   * Delete scheduled and undelivered mail
+   * Deletes all scheduled and undelivered mail from the domain queue.  This endpoint must be called on the storage API host and in the domain's region.  e.g. https://storage-us-east4.api.mailgun.net/v3/example.com/envelopes
+   *
+   * The storage hosts are `storage-us-east4.api.mailgun.net`, `storage-us-west1.api.mailgun.net`, and `storage-europe-west1.api.mailgun.net`.
+   * @param { string } domainName - The name of the domain you want to delete envelope from
+   * @param { MailgunStorageHostname } storageHostname - The storage hostname to be purged
+   */
+  public async purgeSendingQueue(
+    domainName: string,
+    storageHostname: MailgunStorageHostname,
+  ): Promise<GenericResponse> {
```

Comment:
What do you think about having a `void` return value ?
GenericResponse do not have added value over the HTTP 200 return code, or there is a need to define the expected message within response body to clearly define if operation was fine or not (and the `done` string as expected response is not part of specification)
In this context a simple expectation to throw an expectation if error with a void function seems to simplify the interface with SDK

Reply:
Indeed, this makes sense for a DELETE operation

Reply:
done